internal/controller/dnsrecord_controller.go

```diff
@@ -336,8 +339,16 @@ func (r *DNSRecordReconciler) applyChanges(ctx context.Context, dnsRecord *v1alp
 		// generation has not changed but there are changes.
 		// implies that they were overridden - bump write counter
 		if !generationChanged(dnsRecord) {
-			dnsRecord.Status.WriteCounter++
-			logger.V(3).Info("Changes needed on the same generation of record")
+			if dnsRecord.Status.WriteCounter < WriteCounterLimit {
```

Comment:
do we set any dimension on this metric? Seems like we should set the the dnsRecord name and namespace as labels?

Reply:
As I read it we will increment this for all dns records so only ever 20 writes or am I mistaken

Reply:
yes, only 20 writes per generation of DNS record (frankly, if we have more then 2 writes per gen something is wrong with the DNS provider or, more accurately, something keeps overriding changes)